### ISSUE TITLE
chown: /vault: Operation not permitted

### ISSUE THREAD

**Issue by BentCoder** [hashicorp/docker-vault#150]
Hi,

How do get correct permissions on volumes so that the `config`, `file` and `logs` folders are persistent? Just because of the permissions error, I cannot get `file` and `logs` folders working as expected.

Thanks

**OUTPUT**

```bash
$ docker-compose up --build
Creating test_vault_1 ... done
Attaching to test_vault_1
vault_1  | chown: /vault/config/config.json: Operation not permitted
vault_1  | chown: /vault/config: Operation not permitted
vault_1  | chown: /vault/config: Operation not permitted
vault_1  | Could not chown /vault/config (may not have appropriate permissions)
vault_1  | ==> Vault server configuration:
vault_1  | 
vault_1  |                      Cgo: disabled
vault_1  |               Listener 1: tcp (addr: "0.0.0.0:8200", cluster address: "0.0.0.0:8201", max_request_duration: "1m30s", max_request_size: "33554432", tls: "disabled")
vault_1  |                Log Level: info
vault_1  |                    Mlock: supported: true, enabled: false
vault_1  |                  Storage: file
vault_1  |                  Version: Vault v1.1.2
vault_1  |              Version Sha: 0082501623c0b704b87b1fbc84c2d725994bac54
vault_1  | 
vault_1  | ==> Vault server started! Log data will stream in below:
vault_1  | 
vault_1  | 2019-05-06T10:40:23.726Z [WARN]  no `api_addr` value specified in config or in VAULT_API_ADDR; falling back to detection if possible, but this value should be manually set
```

**The current setup**

```bash
.
├── docker-compose.yml
└── vault
    ├── config
    │   └── config.json
    ├── file
    └── logs
```

**docker-compose.yml**
```yml
version: '3'

services:
  vault:
    image: vault:1.1.2
    environment:
      - VAULT_ADDR=http://127.0.0.1:8200
      - SKIP_SETCAP=true
    ports:
      - 8200:8200
    cap_add:
      - IPC_LOCK
    command: server
    volumes:
      - ./vault:/vault
```

**config.json**
```json
{
  "storage": {
    "file": {
      "path": "vault/file"
    }
  },
  "listener": {
    "tcp": {
      "address": "0.0.0.0:8200",
      "tls_disable": true
    }
  },
  "ui": true,
  "max_lease_ttl": "8760h",
  "default_lease_ttl": "8760h",
  "disable_mlock": true
}
```

**docker info**

```bash
docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 2
Server Version: 18.06.1-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 6
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 468a545b9edcd5932818eb9de8e72413e616e86e
runc version: 69663f0bd4b60df09991c08812a60108003fa340
init version: fec3683
Kernel Version: 3.16.0-4-amd64
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.963GiB
Name: docker
ID: JGLA:XDOP:3YRK:F3XU:EWS3:LCWY:UTEL:4C3E:4X47:URSY:HCNI:T4PG
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: hello
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No memory limit support
WARNING: No swap limit support
WARNING: No kernel memory limit support
WARNING: No oom kill disable support
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
```

**Comment by jasonodonnell** [comment]
Hi @BentCoder!

We recently merged #129 which makes the `chown` skippable by setting `SKIP_CHOWN=true`.  Hope that helps!

**Comment by ppoliani** [comment]
@jasonodonnell I added the `SKIP_CHOWN` variable running the docker container but I still get the same error.  I've also created a persistent volume /mnt/vault:/vault. Do I need to add the env variable inside the running container?
